sql/src/main/java/io/crate/execution/dml/SysUpdateResultSetProjector.java

```diff
@@ -0,0 +1,105 @@
+import io.crate.data.Input;
+import io.crate.data.Projector;
+import io.crate.data.Row;
+import io.crate.execution.engine.collect.NestableCollectExpression;
+import io.crate.expression.InputFactory;
+import io.crate.expression.reference.StaticTableReferenceResolver;
+import io.crate.expression.reference.sys.SysRowUpdater;
+import io.crate.expression.reference.sys.check.node.SysNodeCheck;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.Functions;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.sys.SysNodeChecksTableInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Collector;
+
+public class SysUpdateResultSetProjector implements Projector {
+
+    private final Consumer<Object> rowWriter;
+    private final List<NestableCollectExpression<SysNodeCheck, ?>> expressions;
+    private final List<Input<?>> inputs;
+    private final SysRowUpdater<?> sysRowUpdater;
+
+    public SysUpdateResultSetProjector(SysRowUpdater<?> sysRowUpdater,
+                                       Consumer<Object> rowWriter,
+                                       Symbol[] returnValues,
+                                       Functions functions,
+                                       TransactionContext txnCtx) {
+        this.sysRowUpdater = sysRowUpdater;
+        this.rowWriter = rowWriter;
+        InputFactory.Context<NestableCollectExpression<SysNodeCheck, ?>> cntx = new InputFactory(
+            functions).ctxForRefs(
+            txnCtx, new StaticTableReferenceResolver<>(SysNodeChecksTableInfo.expressions()));
+        cntx.add(List.of(returnValues));
+        this.expressions = cntx.expressions();
+        this.inputs = cntx.topLevelInputs();
+    }
+
+    @Override
+    public BatchIterator<Row> apply(BatchIterator<Row> batchIterator) {
+        return CollectingBatchIterator
+            .newInstance(batchIterator,
+                         Collector.of(
+                             ArrayList<Object[]>::new,
+                             (acc, row) -> {
+                                 Object[] returnValues = evaluateReturnValues(row);
+                                 acc.add(returnValues);
+                             },
+                             (state1, state2) -> {
+                                 throw new UnsupportedOperationException(
+                                     "Combine not supported");
+                             },
+                             CollectionBucket::new
+                             ));
+    }
+
+    private Object[] evaluateReturnValues(Row row) {
+        Object sysNodeCheckId = row.get(0);
+        //Update sysNodeCheck to the new value
+        rowWriter.accept(sysNodeCheckId);
+        //Retrieve updated sysNodeCheck and evaluate return values
+        SysNodeCheck sysNodeCheck = (SysNodeCheck) sysRowUpdater.getRow(sysNodeCheckId);
+        expressions.forEach(x -> x.setNextRow(sysNodeCheck));
```

Comment:
Btw, if the lambda captures a variable from the outer scope it causes an allocation.
Shouldn't be a problem for updates on system tables, but something to keep in mind for logic that runs in an inner loop.